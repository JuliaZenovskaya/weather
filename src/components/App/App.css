.App {
  width: 80%;
  margin: 2% 10%;
  background-color: rgb(222, 244, 213);
  padding: 1%;
  border: 2px rgb(50, 112, 51) ridge;
}

.header_text {
  float: left;
  width: 70%;
  font-size: 20px;
  font-weight: bold;
}

.geo_weather {
  margin-top: 10px;
  margin-left: 10px;
}
