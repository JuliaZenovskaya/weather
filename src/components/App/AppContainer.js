export function getGeo() {
  if (navigator.geolocation) {
    navigator.geolocation.getCurrentPosition(position => {

    },
    () => {

      alert('Permission denied. Load weather from default coordinates');
    });
  } else {

  }
}
