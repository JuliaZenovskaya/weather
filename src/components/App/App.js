import React from 'react';
import './App.css';
import { connect } from 'react-redux';
import { setCoords, getErrorResponse } from '../../actions/GeoAction';
import { getWeatherByCoords } from '../../fetch/GetWeatherByCoords';
import Weather from '../Weather/Weather';

class App extends React.Component {
  componentDidMount() {
    this.getGeo();
  }

  render() {
    return (
      <div className="App">
      <div className="head">
        <div className="headText">Погода здесь</div>
        <button className="headButton" onClick={this.getGeo()}>Обновить геолокацию</button>
      </div>
      {this.props.weather && <Weather weather={this.props.weather}/>}
      {this.props.error && <div className="error">Error: {this.props.error}</div>}
      {!this.props.weather && !this.props.error && <div>Данные загружаются...</div>}
      </div>
    );
  }

  getGeo() {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(position => {
        const coords = {
          lat: position.coords.latitude,
          lon: position.coords.longitude
        }
        this.props.setCoords(coords);
        this.props.getWeatherByCoords(this.props.coords);
      },
      () => {
        const coords = {
          lat: 10,
          lon: 10
        }
        this.props.setCoords(coords);
        this.props.getWeatherByCoords(this.props.coords);
      });
    } else {
      this.props.getErrorResponse('your browser does not support geolocation');
    }
  }
}


function mapStateToProps(state) {
return {
  coords: state.geo.coords,
  weather: state.geo.weather,
  error: state.geo.error
};
}


function mapDispatchToProps(dispatch) {
return {
  setCoords: (coords) => {
    dispatch(setCoords(coords));
  },

  getWeatherByCoords: (coords) => {
    dispatch(getWeatherByCoords(coords));
  },

  getErrorResponse: (error) => {
    dispatch(getErrorResponse(error));
  }
};
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
