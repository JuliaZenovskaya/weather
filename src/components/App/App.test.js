import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

const response = {
    "coord":{"lon":37.62,"lat":55.75},
    "weather":[{"id":701,"main":"Mist","description":"mist","icon":"50n"}],
    "base":"stations",
    "main":{"temp":4.38,"pressure":1015,"humidity":100,"temp_min":2,"temp_max":5.56},
    "visibility":3000,
    "wind":{"speed":4,"deg":90},
    "clouds":{"all":90},
    "dt":1573170827,
    "sys":{"type":1,"id":9027,"country":"RU","sunrise":1573188558,"sunset":1573220232},
    "timezone":10800,
    "id":524901,
    "name":"Moscow",
    "cod":200
  };

  it("")
