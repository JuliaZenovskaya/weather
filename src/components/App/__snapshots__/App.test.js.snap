// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather in displaying with favourite cities Some error with displaying weather 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div
      className="error"
    >
      Error: 
      error
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    >
      <div
        className="weather"
      >
        <div
          className="left"
        >
          <div>
            Moscow
          </div>
          <img
            src="https://openweathermap.org/img/wn/50n.png"
          />
          <div>
            -269
             °C
          </div>
          <button
            className="button"
            onClick={[Function]}
          >
            Удалить
          </button>
        </div>
        <div
          className="right"
        >
          <div
            className="pressure"
          >
            Давление: 
            1015
             hPa
          </div>
          <div
            className="humidity"
          >
            Влажность: 
            100
            %
          </div>
          <div
            className="clouds"
          >
            Облачность: 
            mist
          </div>
          <div
            className="wind"
          >
            Ветер: 
            4
             м/с
          </div>
          <div
            className="coords"
          >
            Координаты: [
            37.62
            , 
            55.75
            ]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Weather in displaying with favourite cities Weather exists and is not loading 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div
      className="weather"
    >
      <div
        className="left"
      >
        <div>
          Moscow
        </div>
        <img
          src="https://openweathermap.org/img/wn/50n.png"
        />
        <div>
          -269
           °C
        </div>
      </div>
      <div
        className="right"
      >
        <div
          className="pressure"
        >
          Давление: 
          1015
           hPa
        </div>
        <div
          className="humidity"
        >
          Влажность: 
          100
          %
        </div>
        <div
          className="clouds"
        >
          Облачность: 
          mist
        </div>
        <div
          className="wind"
        >
          Ветер: 
          4
           м/с
        </div>
        <div
          className="coords"
        >
          Координаты: [
          37.62
          , 
          55.75
          ]
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    >
      <div
        className="weather"
      >
        <div
          className="left"
        >
          <div>
            Moscow
          </div>
          <img
            src="https://openweathermap.org/img/wn/50n.png"
          />
          <div>
            -269
             °C
          </div>
          <button
            className="button"
            onClick={[Function]}
          >
            Удалить
          </button>
        </div>
        <div
          className="right"
        >
          <div
            className="pressure"
          >
            Давление: 
            1015
             hPa
          </div>
          <div
            className="humidity"
          >
            Влажность: 
            100
            %
          </div>
          <div
            className="clouds"
          >
            Облачность: 
            mist
          </div>
          <div
            className="wind"
          >
            Ветер: 
            4
             м/с
          </div>
          <div
            className="coords"
          >
            Координаты: [
            37.62
            , 
            55.75
            ]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Weather in displaying with favourite cities Weather exists but is loading 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div>
      Загрузка...
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    >
      <div
        className="weather"
      >
        <div
          className="left"
        >
          <div>
            Moscow
          </div>
          <img
            src="https://openweathermap.org/img/wn/50n.png"
          />
          <div>
            -269
             °C
          </div>
          <button
            className="button"
            onClick={[Function]}
          >
            Удалить
          </button>
        </div>
        <div
          className="right"
        >
          <div
            className="pressure"
          >
            Давление: 
            1015
             hPa
          </div>
          <div
            className="humidity"
          >
            Влажность: 
            100
            %
          </div>
          <div
            className="clouds"
          >
            Облачность: 
            mist
          </div>
          <div
            className="wind"
          >
            Ветер: 
            4
             м/с
          </div>
          <div
            className="coords"
          >
            Координаты: [
            37.62
            , 
            55.75
            ]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Weather in displaying with favourite cities Weather is not exists and is loading 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div>
      Загрузка...
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    >
      <div
        className="weather"
      >
        <div
          className="left"
        >
          <div>
            Moscow
          </div>
          <img
            src="https://openweathermap.org/img/wn/50n.png"
          />
          <div>
            -269
             °C
          </div>
          <button
            className="button"
            onClick={[Function]}
          >
            Удалить
          </button>
        </div>
        <div
          className="right"
        >
          <div
            className="pressure"
          >
            Давление: 
            1015
             hPa
          </div>
          <div
            className="humidity"
          >
            Влажность: 
            100
            %
          </div>
          <div
            className="clouds"
          >
            Облачность: 
            mist
          </div>
          <div
            className="wind"
          >
            Ветер: 
            4
             м/с
          </div>
          <div
            className="coords"
          >
            Координаты: [
            37.62
            , 
            55.75
            ]
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Weather in displaying without favourite cities Some error with displaying weather 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div
      className="error"
    >
      Error: 
      error
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    />
  </div>
</div>
`;

exports[`Weather in displaying without favourite cities Weather exists and is not loading 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div
      className="weather"
    >
      <div
        className="left"
      >
        <div>
          Moscow
        </div>
        <img
          src="https://openweathermap.org/img/wn/50n.png"
        />
        <div>
          -269
           °C
        </div>
      </div>
      <div
        className="right"
      >
        <div
          className="pressure"
        >
          Давление: 
          1015
           hPa
        </div>
        <div
          className="humidity"
        >
          Влажность: 
          100
          %
        </div>
        <div
          className="clouds"
        >
          Облачность: 
          mist
        </div>
        <div
          className="wind"
        >
          Ветер: 
          4
           м/с
        </div>
        <div
          className="coords"
        >
          Координаты: [
          37.62
          , 
          55.75
          ]
        </div>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    />
  </div>
</div>
`;

exports[`Weather in displaying without favourite cities Weather exists but is loading 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div>
      Загрузка...
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    />
  </div>
</div>
`;

exports[`Weather in displaying without favourite cities Weather is not exists and is loading 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header_text"
    >
      Погода здесь
    </div>
    <button
      className="header_button"
      onClick={[Function]}
    >
      Обновить местоположение
    </button>
  </div>
  <div
    className="geo_weather"
  >
    <div>
      Загрузка...
    </div>
  </div>
  <div>
    <div>
      <div
        className="fav_cities"
      >
        Избранное
      </div>
      <form
        onSubmit={[Function]}
      >
        <input
          className="city_input"
          name="city"
          required={true}
          type="text"
        />
        <input
          className="city_button"
          type="submit"
          value="Добавить"
        />
      </form>
    </div>
    <div
      className="weather"
    />
  </div>
</div>
`;
